### PR TITLE
Redux toolkit implemented with test counter slice

### LINKED ISSUE #5: Install and set up Redux Toolkit stores and slices for 'starter' state
# State Management

We need to manage state for this app centrally and easily access that state via custom hooks (hooks to be dealt with in separate tickets).

## Requirements

Use this as a guide: [https://redux-toolkit.js.org/usage/nextjs](https://redux-toolkit.js.org/usage/nextjs)

- Set up simple state for holding all breeds for now (we will paginate down the line).
- Stores and slices should be organised in a neat file structure, follow the docs for best practice.

### PR DESCRIPTION
**Pull request by skoly1**

### CODE REVIEW

File: app/page.tsx [hunk 1]
```diff
@@ -1,16 +1,29 @@
...
     const { setTheme } = useTheme()
+    const state = useAppSelector((state) => state)
+    const dispatch = useAppDispatch()
 
     return (
-        <main className='bg-primar flex min-h-screen justify-around p-24'>
-            <h1>DogDB Frontend</h1>
-            <Button onClick={() => setTheme('light')}>Light Mode</Button>
-            <Button onClick={() => setTheme('dark')}>Dark Mode</Button>
+        <main className='bg-primar min-h-screen p-24'>
```
Comment (adamrichardturner) [inline app/page.tsx:14]: Can we update to primary instead of primar

File: components/StoreProvider/StoreProvider.tsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import { AppStore, makeStore } from '@/store/store'
+import { useRef } from 'react'
+import { Provider } from 'react-redux'
+
+export default function StoreProvider({
+    children,
+}: {
+    children: React.ReactNode
+}) {
+    const storeRef = useRef<AppStore>()
```
Comment (adamrichardturner) [inline components/StoreProvider/StoreProvider.tsx:11]: Interesting approach, but why did we do this?

**Review by adamrichardturner — CHANGES_REQUESTED** [review]
Looks good, though please update the class name for primary

File: components/StoreProvider/StoreProvider.tsx (hunk 2 above)
Comment (skoly1) [reply components/StoreProvider/StoreProvider.tsx:11]: This is to separate server component and client component in next js. As store provider only works in use client, we create a HOC and wrap this in server layout.tsx

File: app/page.tsx (hunk 1 above)
Comment (skoly1) [reply app/page.tsx:14]: Done

**Review by adamrichardturner — APPROVED** [review]
LGTM!

**Pull request merged**